packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -1076,6 +1076,28 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with Restoratio
       },
     );
   }
+  // AutofillClient implementation start.
+  @override
+  String get autofillId => _editableText.autofillId;
+
+  @override
+  void autofill(TextEditingValue newEditingValue) => _editableText.autofill(newEditingValue);
+
+  @override
+  TextInputConfiguration get textInputConfiguration {
+    final List<String>? autofillHints = widget.autofillHints?.toList(growable: false);
+    final AutofillConfiguration autofillConfiguration = autofillHints != null
+      ? AutofillConfiguration(
+          uniqueIdentifier: autofillId,
+          autofillHints: autofillHints,
+          currentEditingValue: _effectiveController.value,
+          hintText: widget.placeholder,
```

Comment:
What is hintText actually used for, I guess somewhere in the engine already?  I understand autofillHints is the main list of hints that's used by the platform to determine what to autofill.

Reply:
AFAIK currently we don't send the hint text. The reason the text input plugin needs it is when the autofill hints list is empty it's a good indication for what the text field is actually for: https://github.com/flutter/flutter/issues/85554